decoimpact/business/entities/rules/depth_average_rule.py

```diff
@@ -0,0 +1,101 @@
+# This file is part of D-EcoImpact
+# Copyright (C) 2022-2024 Stichting Deltares
+# This program is free software distributed under the
+# GNU Affero General Public License version 3.0
+# A copy of the GNU Affero General Public License can be found at
+# https://github.com/Deltares/D-EcoImpact/blob/main/LICENSE.md
+"""
+Module for DepthAverageRule class
+
+Classes:
+    DepthAverageRule
+"""
+from typing import Dict
+
+import xarray as _xr
+
+from decoimpact.business.entities.rules.i_multi_array_based_rule import (
+    IMultiArrayBasedRule,
+)
+from decoimpact.business.entities.rules.rule_base import RuleBase
+from decoimpact.crosscutting.i_logger import ILogger
+
+
+class DepthAverageRule(RuleBase, IMultiArrayBasedRule):
+    """Implementation for the depthaverage rule"""
+
+    def execute(
+        self, value_arrays: Dict[str, _xr.DataArray], logger: ILogger
+    ) -> _xr.DataArray:
+        """Calculate depth average of assumed z-layers.
+        Args:
+            value_array (DataArray): Values to multiply
+        Returns:
+            DataArray: Averaged values
+        """
+
+        # The first DataArray in our value_arrays contains the values to be averaged
+        # But the name of the key is given by the user, so just take the first
+        variable_key = next(iter(value_arrays.keys()))
+        variables = value_arrays[variable_key]
+
+        # depths interfaces = borders of the layers in terms of depth
+        depths_interfaces = value_arrays["mesh2d_interface_z"]
```

Comment:
we now have these names hardcoded in 2 places (here and the parser).
Is it an idea to use a variable for this
value_arrays[interface_name]
, and only hard code the name in a different file (e.g. delft3d_specific_data.py). That makes it easy to later connect to other data sources